lib/bundled_gems.rb

```diff
@@ -105,11 +105,14 @@ def self.warning?(name, specs: nil)
       # and `require "syslog"` to `require "#{ARCHDIR}/syslog.so"`.
       feature.delete_prefix!(ARCHDIR)
       feature.delete_prefix!(LIBDIR)
-      segments = feature.split("/")
+      # 1. A segment for the basic EXACT and SINCE check e.g. `logger`
+      # 2. A segment for the multi-part SINCE check e.g. `resolv-replace`
```

Comment:
I don't think that  resolv-replace, which provides `resolv-replace.rb` as the name says, is the targets of this code.
The targets are net/* family, net-http gem for `require "net/http"`.

I suspect that this code can be removed since such libraries are no longer in `SINCE` table.

Reply:
I've gone ahead and removed them.

Reply:
@hsbt What do you think? Or we should keep the feature?

Reply:
I don't want to remove these tests.

Reply:
In my opinion, it's better to not keep code not necessary in real life, but just meant to make some specs pass, so I would remove the code and tests as suggested.

If we want to make sure we don't regress if gems with the special naming are ever added back to the `SINCE` list, we could make sure the specs cover the real `SINCE` list more thoroughly, and that they start failing if a gem with the special naming is added in the future.